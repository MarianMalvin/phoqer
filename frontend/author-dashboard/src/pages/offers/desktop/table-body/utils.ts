import { TableCellSizeEnum } from 'phoqer';

export interface HeaderType {
    key: string;
    title: string;
}

export const headerOrder: HeaderType[] = [
    { key: 'status', title: 'Status' },
    { key: 'options', title: 'Options' },
    { key: 'images', title: 'Images' },
    { key: 'title', title: 'Title' },
    { key: 'category', title: 'Category' },
    { key: 'reviews', title: 'Reviews' },
    { key: 'price', title: 'Price' },
    { key: 'sale_percentage', title: 'Sale percentage' },
    { key: 'sale_description', title: 'Sale description' },
    { key: 'description', title: 'Description' },
];

export const sizeMap: { [key: string]: TableCellSizeEnum } = {
    status: TableCellSizeEnum.Small,
    options: TableCellSizeEnum.Middle,
    images: TableCellSizeEnum.Small2Extra,
    title: TableCellSizeEnum.Large5Extra,
    description: TableCellSizeEnum.Large8Extra,
    sale_percentage: TableCellSizeEnum.MiddleExtra,
    sale_description: TableCellSizeEnum.Large4Extra,
    reviews: TableCellSizeEnum.Middle,
    price: TableCellSizeEnum.Middle,
    category: TableCellSizeEnum.Middle,
};
