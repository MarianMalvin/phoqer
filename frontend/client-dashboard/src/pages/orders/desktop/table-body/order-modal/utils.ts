export const cellMap = [
    { key: 'title', title: 'Title' },
    { key: 'status', title: 'Status' },
    { key: 'date', title: 'Created' },
    { key: 'expired', title: 'Expired' },
    { key: 'price', title: 'Price' },
    { key: 'category', title: 'Category' },
    { key: 'country', title: 'Country' },
    { key: 'city', title: 'City' },
    { key: 'zip', title: 'ZIP' },
    { key: 'address', title: 'Address' },
    { key: 'comment', title: 'Comment' },
    { key: 'sale', title: 'Sale' },
    { key: 'sale_description', title: 'Sale Description' },
];
