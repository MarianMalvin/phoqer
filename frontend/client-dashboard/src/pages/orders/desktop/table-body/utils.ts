import { TableCellSizeEnum } from 'phoqer';

export interface HeaderType {
    key: string;
    title: string;
}

export const headerOrder: HeaderType[] = [
    { key: 'status', title: 'Status' },
    { key: 'options', title: 'Options' },
    { key: 'images', title: 'Images' },
    { key: 'title', title: 'Title' },
    { key: 'price', title: 'Price' },
    { key: 'user', title: 'Author' },
    { key: 'address', title: 'Address' },
    { key: 'comment', title: 'Comment' },
    { key: 'date', title: 'Created' },
    { key: 'expired', title: 'Expiration Date' },
];

export const sizeMap: { [key: string]: TableCellSizeEnum } = {
    status: TableCellSizeEnum.Small,
    options: TableCellSizeEnum.Middle,
    title: TableCellSizeEnum.Large3Extra,
    images: TableCellSizeEnum.Small2Extra,
    price: TableCellSizeEnum.Middle,
    address: TableCellSizeEnum.Large4Extra,
    comment: TableCellSizeEnum.Large3Extra,
    user: TableCellSizeEnum.LargeExtra,
    date: TableCellSizeEnum.Large2Extra,
    expired: TableCellSizeEnum.Large2Extra,
};
