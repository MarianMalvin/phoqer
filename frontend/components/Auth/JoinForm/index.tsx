import { faEye, faEyeSlash } from '@fortawesome/free-regular-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import Link from 'next/link';
import React, { FormEvent, ReactElement, useState } from 'react';
import { createUseStyles } from 'react-jss';

import { Theme } from '../../../config/theme';

const useStyles = createUseStyles((theme: Theme) => ({
  title: {
    fontSize: theme.rem(1.8),
    fontWeight: theme.text.weight[3],
    textAlign: 'center',
  },
  wrp: {
    display: 'block',
    margin: theme.rem(2, 0),
  },
  inner: {
    position: 'relative',
  },
  label: {
    margin: 0,
    fontSize: theme.rem(1.2),
    fontWeight: theme.text.weight[3],
  },
  input: {
    display: 'block',
    width: '100%',
    height: theme.rem(5),
    padding: theme.rem(1),
    paddingRight: theme.rem(5),
    background: theme.palette.blue[2],
    border: theme.border(0.1, theme.palette.gray[3]),
    borderRadius: theme.radius,
    fontSize: theme.rem(1.4),
    lineHeight: 1,
    outline: 'none',
  },
  eye: {
    position: 'absolute',
    top: '52%',
    right: 0,
    height: '100%',
    width: theme.rem(5),
    transform: 'translateY(-50%)',
    fontSize: theme.rem(1.2),
    color: theme.palette.gray[3],
  },
  pass: {
    display: 'block',
    textAlign: 'right',
    fontSize: theme.rem(1.4),
    color: theme.palette.blue[0],
  },
  btn: {
    display: 'block',
    margin: '3rem auto 2rem',
    padding: theme.rem(1.5, 2),
    fontWeight: theme.text.weight[4],
    textAlign: 'center',
    fontSize: theme.rem(1.4),
    borderRadius: theme.radius,
    background: theme.palette.grayblue[0],
  },
  text: {
    marginBottom: theme.rem(2),
    fontSize: theme.rem(1.4),
    fontWeight: theme.text.weight[2],
    textAlign: 'center',
  },
  list: {
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
  },
  socials: {
    height: theme.rem(6),
    width: theme.rem(6),
    margin: theme.rem(1),
    padding: theme.rem(1),
    borderRadius: '50%',
    background: theme.palette.gray[1],
  },
  facebook: {
    background: theme.palette.blue[0],
  },
  google: {
    background: theme.palette.red[0],
  },
  svg: {
    height: theme.rem(2),
    width: theme.rem(1.5),
    fill: theme.palette.white,
  },
}));

const LoginForm = (): ReactElement => {
  const css = useStyles();
  const [password, setPassword] = useState(true);

  const hadleClick = () => {
    setPassword(!password);
  };

  const handleSubmit = (event: FormEvent) => {
    event.preventDefault();
    alert('fuck you!');
  };

  return (
    <form action="#" method="post" onSubmit={handleSubmit}>
      <h2 className={css.title}>Бобро пожаловать, засранец.</h2>

      <label className={css.wrp}>
        <p className={css.label}>Full Name</p>
        <input type="text" name="full_name" className={css.input} />
      </label>

      <label className={css.wrp}>
        <p className={css.label}>Email</p>
        <input type="email" name="email" className={css.input} />
      </label>

      <label className={css.wrp}>
        <p className={css.label}>Password</p>

        <div className={css.inner}>
          <button className={css.eye} onClick={hadleClick} type="button">
            {password ? (
              <FontAwesomeIcon icon={faEyeSlash} />
            ) : (
              <FontAwesomeIcon icon={faEye} />
            )}
          </button>
          <input
            type={password ? 'password' : 'text'}
            name="password"
            className={css.input}
          />
        </div>
      </label>

      <Link href="/forgot_pass">
        <a className={css.pass}>Забыли пароль?</a>
      </Link>

      <button className={css.btn} type="submit">
        ЗАРЕГИСТРИРОВАТЬСЯ
      </button>

      <p className={css.text}>
        или
        <br />
        залогинтесь с
      </p>

      <ul className={css.list}>
        <li>
          <button
            className={`${css.socials} ${css.facebook}`}
            type="button"
            onClick={handleSubmit}
          >
            <svg
              className={css.svg}
              xmlns="http://www.w3.org/2000/svg"
              xmlnsXlink="http://www.w3.org/1999/xlink"
              version="1.1"
              x="0px"
              y="0px"
              viewBox="0 0 512 512"
              xmlSpace="preserve"
            >
              <path d="M288,176v-64c0-17.664,14.336-32,32-32h32V0h-64c-53.024,0-96,42.976-96,96v80h-64v80h64v256h96V256h64l32-80H288z" />
            </svg>
          </button>
        </li>
        <li>
          <button
            className={`${css.socials} ${css.google}`}
            type="button"
            onClick={handleSubmit}
          >
            <svg
              className={css.svg}
              xmlns="http://www.w3.org/2000/svg"
              xmlnsXlink="http://www.w3.org/1999/xlink"
              version="1.1"
              x="0px"
              y="0px"
              viewBox="0 0 510 510"
              xmlSpace="preserve"
            >
              <path d="M286.875,229.5v63.75h150.45c-15.3,89.25-86.7,153-175.95,153c-104.55,0-191.25-86.7-191.25-191.25    s86.7-191.25,191.25-191.25c53.55,0,99.45,22.95,132.6,58.65l45.9-45.9c-45.9-45.9-107.1-76.5-178.5-76.5    c-140.25,0-255,114.75-255,255s114.75,255,255,255s242.25-114.75,242.25-255v-25.5H286.875z" />
            </svg>
          </button>
        </li>
        <li>
          <button
            className={`${css.socials}`}
            type="button"
            onClick={handleSubmit}
          ></button>
        </li>
      </ul>
    </form>
  );
};

export default LoginForm;
