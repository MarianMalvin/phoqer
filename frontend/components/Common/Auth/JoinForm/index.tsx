import { faEye } from '@fortawesome/free-regular-svg-icons/faEye';
import { faEyeSlash } from '@fortawesome/free-regular-svg-icons/faEyeSlash';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import React, { FormEvent, MouseEvent, ReactElement, useState } from 'react';
import { createUseStyles } from 'react-jss';

import { Theme } from '../../../../assets/theme';
import GoogleFacebook from '../../GoogleFacebook';
import Spinner from '../../Preloaders/Spinner';

const useStyles = createUseStyles((theme: Theme) => ({
    title: {
        fontSize: theme.rem(1.8),
        fontWeight: theme.text.weight[3],
        color: theme.palette.black[0],
        textAlign: 'center',

        '@media (max-width: 500px)': {
            fontSize: theme.rem(2.4),
        },
    },
    wrp: {
        display: 'block',
        margin: theme.rem(2, 0),
    },
    inner: {
        position: 'relative',
    },
    label: {
        margin: 0,
        fontSize: theme.rem(1.2),
        fontWeight: theme.text.weight[3],
        color: theme.palette.black[0],

        '@media (max-width: 500px)': {
            fontSize: theme.rem(1.6),
        },
    },
    input: {
        ...theme.input,
        padding: theme.rem(0, 5, 0, 2),
        background: theme.palette.gray[1],
        color: theme.palette.black[0],

        '@media (max-width: 500px)': {
            fontSize: theme.rem(1.6),
        },
    },
    eye: {
        position: 'absolute',
        top: '52%',
        right: 0,
        height: '100%',
        width: theme.rem(5),
        transform: 'translateY(-50%)',
        fontSize: theme.rem(1.2),
        color: theme.palette.gray[4],

        '& svg': {
            height: theme.rem(2),
            width: theme.rem(2),
        },
    },
    pass: {
        display: 'block',
        textAlign: 'right',
        fontSize: theme.rem(1.4),
        color: theme.palette.primary[0],
    },
    btn: {
        display: 'block',
        minWidth: theme.rem(20),
        margin: '3rem auto 2rem',
        padding: theme.rem(1.5, 2),
        fontWeight: theme.text.weight[4],
        textAlign: 'center',
        fontSize: theme.rem(1.4),
        borderRadius: theme.radius,
        background: theme.palette.primary[0],
        color: theme.palette.trueWhite,

        '& svg': {
            height: theme.rem(1.4),
            width: theme.rem(1.4),
        },

        '@media (max-width: 500px)': {
            margin: '5rem auto 3rem',
            padding: theme.rem(2),
        },
    },
    text: {
        marginBottom: theme.rem(1),
        fontSize: theme.rem(1.4),
        fontWeight: theme.text.weight[2],
        color: theme.palette.black[0],
        textAlign: 'center',

        '@media (max-width: 500px)': {
            marginBottom: theme.rem(2),
        },
    },
    svg: {
        height: theme.rem(2),
        width: theme.rem(1.5),
        fill: 'inherit',
    },
}));

const JoinForm = (): ReactElement => {
    const css = useStyles();
    const [loading, setLoading] = useState(false);
    const [password, setPassword] = useState(true);

    const handleClick = () => {
        setPassword(!password);
    };

    const handleSubmit = (event: FormEvent<HTMLFormElement> | MouseEvent<HTMLButtonElement>) => {
        event.preventDefault();
        setLoading(true);
        alert('fuck you!');
    };

    return (
        <form action="#" method="post" onSubmit={handleSubmit}>
            <h2 className={css.title}>Бобро пожаловать, засранец.</h2>

            <label className={css.wrp}>
                <p className={css.label}>Full Name</p>
                <input type="text" name="full_name" className={css.input} />
            </label>

            <label className={css.wrp}>
                <p className={css.label}>Email</p>
                <input type="email" name="email" className={css.input} />
            </label>

            <label className={css.wrp}>
                <p className={css.label}>Password</p>

                <div className={css.inner}>
                    <button className={css.eye} onClick={handleClick} type="button">
                        {password ? <FontAwesomeIcon icon={faEyeSlash} /> : <FontAwesomeIcon icon={faEye} />}
                    </button>
                    <input type={password ? 'password' : 'text'} name="password" className={css.input} />
                </div>
            </label>

            <button className={css.btn} style={loading ? { pointerEvents: 'none' } : {}} type="submit">
                {loading ? <Spinner /> : 'ЗАРЕГИСТРИРОВАТЬСЯ'}
            </button>

            <p className={css.text}>
                или
                <br />
                залогинтесь с
            </p>

            <GoogleFacebook />
        </form>
    );
};

export default JoinForm;
