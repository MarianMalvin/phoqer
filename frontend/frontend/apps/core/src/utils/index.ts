export * from './recentOffers';
export * from './errorHandler';
