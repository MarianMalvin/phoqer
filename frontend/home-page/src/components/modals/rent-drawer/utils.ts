import { RentPayload } from './type';

export const RENT_DRAWER_DATA_KEY = 'rent-drawer-data';

export const initialValues: RentPayload = {
    country: '',
    city: '',
    zip: '',
    address: '',
    comment: '',
};

export const getInitialValues = (): RentPayload => {
    try {
        const json = localStorage.getItem(RENT_DRAWER_DATA_KEY);
        if (!json) {
            return initialValues;
        }

        const data = JSON.parse(json);
        return data || initialValues;
    } catch {
        return initialValues;
    }
};

export const setInitialValues = (value: RentPayload): void => {
    try {
        localStorage.setItem(RENT_DRAWER_DATA_KEY, JSON.stringify(value));
    } catch (error) {
        console.log(error);
    }
};
