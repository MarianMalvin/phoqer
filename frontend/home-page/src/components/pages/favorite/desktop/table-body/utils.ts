export interface HeaderType {
    key: string;
    title: string;
}

export const headerOrder: HeaderType[] = [
    { key: 'image', title: 'Image' },
    { key: 'options', title: 'Rent now' },
    { key: 'title', title: 'Title' },
    { key: 'price', title: 'Price' },
    { key: 'sale', title: 'Sale' },
    { key: 'category', title: 'Category' },
];

export const sizeMap: { [key: string]: number } = {
    image: 10,
    title: 45,
    price: 15,
    sale: 10,
    category: 12,
    options: 18,
};
