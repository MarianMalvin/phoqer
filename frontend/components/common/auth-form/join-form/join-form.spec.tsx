import { render } from '../../../../tests/tests.utils';
import JoinForm from './index';

describe('JoinForm component', () => {
    it('matches snapshot', () => {
        const { asFragment } = render(<JoinForm />);
        expect(asFragment()).toMatchSnapshot();
    });
});
