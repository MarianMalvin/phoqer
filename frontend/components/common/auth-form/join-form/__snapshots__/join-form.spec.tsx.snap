// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JoinForm component matches snapshot 1`] = `
<DocumentFragment>
  <form
    action="#"
    method="post"
  >
    <h2
      class="title-0-2-1"
    >
      Welcome!
    </h2>
    <label
      class="wrp-0-2-2"
    >
      <p
        class="label-0-2-4"
      >
        First name
      </p>
      <div
        class="wrp-0-2-13"
      >
        <div
          class="inner-0-2-14"
        >
          <input
            autocomplete="given-name"
            class="input-0-2-15"
            name="first_name"
            placeholder="First name"
            type="text"
            value=""
          />
        </div>
      </div>
    </label>
    <label
      class="wrp-0-2-2"
    >
      <p
        class="label-0-2-4"
      >
        Last name
      </p>
      <div
        class="wrp-0-2-13"
      >
        <div
          class="inner-0-2-14"
        >
          <input
            autocomplete="family-name"
            class="input-0-2-15"
            name="last_name"
            placeholder="Last name"
            type="text"
            value=""
          />
        </div>
      </div>
    </label>
    <label
      class="wrp-0-2-2"
    >
      <p
        class="label-0-2-4"
      >
        Email
      </p>
      <div
        class="wrp-0-2-13"
      >
        <div
          class="inner-0-2-14"
        >
          <input
            autocomplete="email"
            class="input-0-2-15"
            name="email"
            placeholder="Email"
            type="email"
            value=""
          />
        </div>
      </div>
    </label>
    <label
      class="wrp-0-2-2"
    >
      <p
        class="label-0-2-4"
      >
        Password
      </p>
      <div
        class="inner-0-2-3"
      >
        <div
          class="wrp-0-2-22"
        >
          <div
            class="tooltip-0-2-23 bottom_right-0-2-27 tooltip"
          >
            <ul
              class="list-0-2-10"
            >
              <li
                class="red-0-2-11"
              >
                Minimum 8 characters
              </li>
              <li
                class="red-0-2-11"
              >
                Numbers and letters
              </li>
              <li
                class="red-0-2-11"
              >
                Uppercase
              </li>
              <li
                class="red-0-2-11"
              >
                Special characters
              </li>
            </ul>
          </div>
          <div
            class="wrp-0-2-13"
          >
            <div
              class="inner-0-2-14"
            >
              <input
                autocomplete="current-password"
                class="input-0-2-15 withPassword-0-2-20"
                name="password"
                placeholder="Password"
                type="password"
                value=""
              />
              <button
                class="icon-0-2-18"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-eye-slash fa-w-20 "
                  data-icon="eye-slash"
                  data-prefix="far"
                  focusable="false"
                  role="img"
                  viewBox="0 0 640 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M634 471L36 3.51A16 16 0 0 0 13.51 6l-10 12.49A16 16 0 0 0 6 41l598 467.49a16 16 0 0 0 22.49-2.49l10-12.49A16 16 0 0 0 634 471zM296.79 146.47l134.79 105.38C429.36 191.91 380.48 144 320 144a112.26 112.26 0 0 0-23.21 2.47zm46.42 219.07L208.42 260.16C210.65 320.09 259.53 368 320 368a113 113 0 0 0 23.21-2.46zM320 112c98.65 0 189.09 55 237.93 144a285.53 285.53 0 0 1-44 60.2l37.74 29.5a333.7 333.7 0 0 0 52.9-75.11 32.35 32.35 0 0 0 0-29.19C550.29 135.59 442.93 64 320 64c-36.7 0-71.71 7-104.63 18.81l46.41 36.29c18.94-4.3 38.34-7.1 58.22-7.1zm0 288c-98.65 0-189.08-55-237.93-144a285.47 285.47 0 0 1 44.05-60.19l-37.74-29.5a333.6 333.6 0 0 0-52.89 75.1 32.35 32.35 0 0 0 0 29.19C89.72 376.41 197.08 448 320 448c36.7 0 71.71-7.05 104.63-18.81l-46.41-36.28C359.28 397.2 339.89 400 320 400z"
                    fill="currentColor"
                  />
                </svg>
              </button>
            </div>
          </div>
        </div>
      </div>
    </label>
    <button
      class="btn-0-2-30 btn-0-2-7"
      type="submit"
    >
      <span>
        Join Phoqer
      </span>
    </button>
    <p
      class="text-0-2-8"
    >
      or
      <br />
      login with
    </p>
    <ul
      class="list-0-2-32"
    >
      <li>
        <button
          class="socials-0-2-33 facebook-0-2-34"
          type="button"
        >
          <svg
            id="Layer_1"
            version="1.1"
            viewBox="0 0 512 512"
            x="0px"
            xml:space="preserve"
            xmlns="http://www.w3.org/2000/svg"
            xmlns:xlink="http://www.w3.org/1999/xlink"
            y="0px"
          >
            <path
              d="M134.941,272.691h56.123v231.051c0,4.562,3.696,8.258,8.258,8.258h95.159  c4.562,0,8.258-3.696,8.258-8.258V273.78h64.519c4.195,0,7.725-3.148,8.204-7.315l9.799-85.061c0.269-2.34-0.472-4.684-2.038-6.44  c-1.567-1.757-3.81-2.763-6.164-2.763h-74.316V118.88c0-16.073,8.654-24.224,25.726-24.224c2.433,0,48.59,0,48.59,0  c4.562,0,8.258-3.698,8.258-8.258V8.319c0-4.562-3.696-8.258-8.258-8.258h-66.965C309.622,0.038,308.573,0,307.027,0  c-11.619,0-52.006,2.281-83.909,31.63c-35.348,32.524-30.434,71.465-29.26,78.217v62.352h-58.918c-4.562,0-8.258,3.696-8.258,8.258  v83.975C126.683,268.993,130.379,272.691,134.941,272.691z"
              style="fill: #ffffff;"
            />
          </svg>
        </button>
      </li>
      <li>
        <button
          class="socials-0-2-33 google-0-2-35"
          type="button"
        >
          <svg
            enable-background="new 0 0 512 512"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g>
              <path
                d="m120 256c0-25.367 6.989-49.13 19.131-69.477v-86.308h-86.308c-34.255 44.488-52.823 98.707-52.823 155.785s18.568 111.297 52.823 155.785h86.308v-86.308c-12.142-20.347-19.131-44.11-19.131-69.477z"
                fill="#fbbd00"
              />
              <path
                d="m256 392-60 60 60 60c57.079 0 111.297-18.568 155.785-52.823v-86.216h-86.216c-20.525 12.186-44.388 19.039-69.569 19.039z"
                fill="#0f9d58"
              />
              <path
                d="m139.131 325.477-86.308 86.308c6.782 8.808 14.167 17.243 22.158 25.235 48.352 48.351 112.639 74.98 181.019 74.98v-120c-49.624 0-93.117-26.72-116.869-66.523z"
                fill="#31aa52"
              />
              <path
                d="m512 256c0-15.575-1.41-31.179-4.192-46.377l-2.251-12.299h-249.557v120h121.452c-11.794 23.461-29.928 42.602-51.884 55.638l86.216 86.216c8.808-6.782 17.243-14.167 25.235-22.158 48.352-48.353 74.981-112.64 74.981-181.02z"
                fill="#3c79e6"
              />
              <path
                d="m352.167 159.833 10.606 10.606 84.853-84.852-10.606-10.606c-48.352-48.352-112.639-74.981-181.02-74.981l-60 60 60 60c36.326 0 70.479 14.146 96.167 39.833z"
                fill="#cf2d48"
              />
              <path
                d="m256 120v-120c-68.38 0-132.667 26.629-181.02 74.98-7.991 7.991-15.376 16.426-22.158 25.235l86.308 86.308c23.753-39.803 67.246-66.523 116.87-66.523z"
                fill="#eb4132"
              />
            </g>
          </svg>
        </button>
      </li>
    </ul>
  </form>
</DocumentFragment>
`;
